packages/hoppscotch-backend/src/shared-request/shared-request.service.ts

```diff
@@ -0,0 +1,217 @@
+    return TO.none;
+  }
+
+  onUserDelete(user: AuthUser): T.Task<void> {
+    return async () => {
+      await this.deleteUserSharedRequests(user.uid);
+    };
+  }
+
+  /**
+   * Delete all the Users SharedRequests
+   * @param uid User Uid
+   * @returns number of all deleted user SharedRequests
+   */
+  async deleteUserSharedRequests(uid: string) {
+    const deletedShortCodes = await this.prisma.shortcode.deleteMany({
+      where: {
+        creatorUid: uid,
+      },
+    });
+
+    return deletedShortCodes.count;
+  }
+
+  /**
+   * Converts a Prisma SharedRequest type into the SharedRequest model
+   *
+   * @param sharedRequestInfo Prisma SharedRequest type
+   * @returns GQL SharedRequest
+   */
+  private cast(sharedRequestInfo: DBSharedRequest): SharedRequest {
+    return <SharedRequest>{
+      id: sharedRequestInfo.id,
+      request: JSON.stringify(sharedRequestInfo.request),
+      properties: JSON.stringify(sharedRequestInfo.properties),
+      createdOn: sharedRequestInfo.createdOn,
+    };
+  }
+  /**
+   * Generate a shortcode
+   *
+   * @returns generated shortcode
+   */
+  private generateShortCodeID(): string {
+    let result = '';
+    for (let i = 0; i < SHORT_CODE_LENGTH; i++) {
+      result +=
+        SHORT_CODE_CHARS[Math.floor(Math.random() * SHORT_CODE_CHARS.length)];
+    }
+    return result;
+  }
+
+  /**
+   * Check to see if ShortCode is already present in DB
+   *
+   * @returns Shortcode
+   */
+  private async generateUniqueShortCodeID() {
+    while (true) {
+      const code = this.generateShortCodeID();
+
+      const data = await this.getSharedRequest(code);
+      if (E.isLeft(data)) return E.right(code);
+    }
+  }
+
+  /**
+   * Fetch details regarding a SharedRequest
+   *
+   * @param sharedRequestID SharedRequest
+   * @returns Either of SharedRequest details or error
+   */
+  async getSharedRequest(sharedRequestID: string) {
+    try {
+      const sharedRequest = await this.prisma.shortcode.findFirstOrThrow({
+        where: { id: sharedRequestID },
+      });
+      return E.right(this.cast(sharedRequest));
+    } catch (error) {
+      return E.left(SHARED_REQUEST_NOT_FOUND);
+    }
+  }
+
+  /**
+   * Create a new SharedRequest
+   *
+   * @param request JSON string of request details
+   * @param properties JSON string of embed properties, if present
+   * @returns Either of SharedRequest or error
+   */
+  async createSharedRequest(
+    request: string,
+    properties: string | null,
+    userInfo: AuthUser,
+  ) {
+    const requestData = stringToJson(request);
+    if (E.isLeft(requestData))
+      return E.left(SHARED_REQUEST_INVALID_REQUEST_JSON);
+
+    let propertiesData;
+    if (!properties) propertiesData = undefined;
+    const parsedProperties = stringToJson(properties);
+    if (E.isLeft(parsedProperties))
+      return E.left(SHARED_REQUEST_INVALID_PROPERTIES_JSON);
+    propertiesData = parsedProperties.right;
```

Comment:
I think this will be more readable. Am I missing any corner case?

see my next review as well to better understand in `this.prisma.shortcode.create`

```suggestion
    const parsedProperties = stringToJson(properties);
    if (E.isLeft(parsedProperties))
      return E.left(SHARED_REQUEST_INVALID_PROPERTIES_JSON);
```